clifford/test/test_jit_func.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+import numpy as np
+import time
+from ..jit_func import jit_func
+
+
+class TestJITFunc(unittest.TestCase):
+
+    def test_reverse(self):
+        from clifford.g3c import layout, blades
+        e12 = blades['e12']
+
+        @jit_func(layout=layout, ast_debug=True)
+        def test_func(A):
+            op = ~A
+            return op
+
+        def slow_test_func(A):
+            op = ~A
+            return op
+
+        np.testing.assert_allclose(test_func(e12).value, slow_test_func(e12).value)
+
+    def test_compound_expression(self):
+        from clifford.g3c import layout, blades
+        e1 = blades['e1']
+        e2 = blades['e2']
+        einf = layout.einf
+
+        @jit_func(layout=layout, ast_debug=True)
+        def test_func(A, B, C):
+            op = ~(((A * B) * C) | (B ^ A)) - 3.1 - A - 7 * B + 5 + C + 2.5 + (2 ^ (A * B * C) ^ 3) + (A | 5)
+            return op
+
+        def slow_test_func(A, B, C):
+            op = ~(((A * B) * C) | (B ^ A)) - 3.1 - A - 7 * B + 5 + C + 2.5 + (2 ^ (A * B * C) ^ 3) + (A | 5)
+            return op
+
+        np.testing.assert_allclose(test_func(e1, e2, einf).value, slow_test_func(e1, e2, einf).value)
+
+    def test_benchmark(self):
+        from clifford.g3c import layout, blades
+        e1 = blades['e1']
+        e2 = blades['e2']
+        einf = layout.einf
+
+        @jit_func(layout=layout)
+        def test_func(A, B, C):
+            op = ~(((A * B) * C) | (B ^ A)) - 3.1 - A - 7 * B + 5 + C + 2.5 + (2 ^ (A * B * C) ^ 3) + (A | 5)
+            return op
+
+        def slow_test_func(A, B, C):
+            op = ~(((A * B) * C) | (B ^ A)) - 3.1 - A - 7 * B + 5 + C + 2.5 + (2 ^ (A * B * C) ^ 3) + (A | 5)
+            return op
```

Comment:
To avoid repeating the function:
```suggestion
        def slow_test_func(A, B, C):
            op = ~(((A * B) * C) | (B ^ A)) - 3.1 - A - 7 * B + 5 + C + 2.5 + (2 ^ (A * B * C) ^ 3) + (A | 5)
            return op
            
        test_func = jit_func(layout)(slow_test_func)
```

Reply:
I had a go at using this syntax and it really wasn't happy, I think calling the decorator in this way messes up the link to the source in some way, didn't dig into it too deeply

Reply:
I suspect it's just you trimming a decorator which is no longer there

Reply:
that's a fair point actually, I guess changing the trim to be in the ast would probably remove that problem, I'll have a look at it